Mathlib/RingTheory/Etale.lean

```diff
@@ -440,19 +440,9 @@ instance FormallyUnramified.base_change [FormallyUnramified R A] :
   intro C _ _ I hI f₁ f₂ e
   letI := ((algebraMap B C).comp (algebraMap R B)).toAlgebra
   haveI : IsScalarTower R B C := IsScalarTower.of_algebraMap_eq' rfl
-  apply AlgHom.restrictScalars_injective R
-  apply TensorProduct.ext
-  intro b a
-  have : b ⊗ₜ[R] a = b • (1 : B) ⊗ₜ a := by rw [TensorProduct.smul_tmul', smul_eq_mul, mul_one]
-  rw [this, AlgHom.restrictScalars_apply, AlgHom.restrictScalars_apply, map_smul, map_smul]
-  congr 1
-  change
-    ((f₁.restrictScalars R).comp TensorProduct.includeRight) a =
-      ((f₂.restrictScalars R).comp TensorProduct.includeRight) a
-  congr 1
-  refine' FormallyUnramified.ext I ⟨2, hI⟩ _
-  intro x
-  exact AlgHom.congr_fun e (1 ⊗ₜ x)
+  ext : 1
+  · exact Subsingleton.elim _ _
+  · exact FormallyUnramified.ext I ⟨2, hI⟩ fun x => AlgHom.congr_fun e (1 ⊗ₜ x)
```

Comment:
Only after writing `Algebra.TensorProduct.ext` did I find out that there was a duplicate implementation hiding here!

Reply:
lgtm